src/vm/siginfo.cpp

```diff
@@ -4953,11 +4953,54 @@ void PromoteCarefully(promote_func   fn,
     (*fn) (ppObj, sc, flags);
 }
 
+void ReportByRefPointersFromByRefLikeObject(promote_func *fn, ScanContext *sc, PTR_MethodTable pMT, PTR_VOID pSrc)
+{
+    WRAPPER_NO_CONTRACT;
+
+    _ASSERTE(pMT->IsByRefLike());
+
+    if (pMT == g_TypedReferenceMT)
```

Comment:
Ok how does 797ec397c5b1db784c137bb4a0938e17be449df7 look? I kept the check considering TypedReference is currently not implemented the way in which you suggest, but the new code treats it similarly to ByReferece in the sense the first field of both is the pointer, which should conform to the way in which TypedReference is currently implemented.